releasenotes/notes/operator-from-circuit-bugfix-5dab5993526a2b0a.yaml

```
@@ -0,0 +1,12 @@
+---
+features_quantum_info:
+  - |
+    Added a new test case to check the initialization of :class:`.Operator` via :meth:`.Operator.from_circuit`
+    when both an initial layout and a final layout is specified in the `_layout` attribute of a :class:`.QuantumCircuit` object.
+    Previously, the final layout was processed without regards for the initial layout, i.e. the
+    initialization was incorrect for all quantum circuits that have a non-trivial initial layout. In
+    addition, this behavior was fixed in :meth:`.Operator.from_circuit`, i.e. an :class:`.Operator` can
+    now be correctly initialized from a quantum circuit with arbitrary initial and final layouts.
+    Lastly, the documentation of the constructor of :class:`.Operator` was adapted to hint to :meth:`.Operator.from_circuit`
+    when e.g. the equivalency of two operators should be determined with non-canonical qubit permutations
+    specified via initial and final layouts.
```

In general the release notes are documentation for Qiskit's end users that gets collated and published on release. So we try to write them from the perspective of informing an existing user of the API about what they need to know about the change included in this PR.  In this case the key detail to document is that you've fixed a bug in the handling of the final layout in the operator construction where in previous releases this was done incorrectly.

```suggestion
fixes:
  - |
    Fixed an issue with the :meth:`.Operator.from_circuit` constructor method where it would incorrectly
    interpret the final layout permutation resulting in an invalid `Operator` being constructed.
    Previously, the final layout was processed without regards for the initial layout, i.e. the
    initialization was incorrect for all quantum circuits that have a non-trivial initial layout.
```